netty/src/main/java/io/grpc/transport/netty/GrpcSslContexts.java

```diff
@@ -93,7 +111,37 @@ public static SslContextBuilder forServer(
    * an application requires particular settings it should override the options set here.
    */
   public static SslContextBuilder configure(SslContextBuilder builder) {
+    // The default client/server providers in Netty are the same, so it doesn't matter which
+    // method we call.
+    ApplicationProtocolConfig protocolConfig =
+            selectApplicationProtocolConfig(SslContext.defaultServerProvider());
     return builder.ciphers(Http2SecurityUtil.CIPHERS, SupportedCipherSuiteFilter.INSTANCE)
-        .applicationProtocolConfig(DEFAULT_APN);
+        .applicationProtocolConfig(protocolConfig);
+  }
+
+  /**
+   * Attempts to select the best {@link ApplicationProtocolConfig} for the given
+   * {@link SslProvider}.
+   */
+  public static ApplicationProtocolConfig selectApplicationProtocolConfig(SslProvider provider) {
+    switch (provider) {
+      case JDK: {
+        if (JettyTlsUtil.isJettyAlpnConfigured()) {
+          return ALPN;
+        }
+        if (JettyTlsUtil.isJettyNpnConfigured()) {
+          return NPN;
+        }
+        throw new IllegalArgumentException("Jetty ALPN/NPN has not been properly configured.");
+      }
+      case OPENSSL:
```

Comment:
I'm saying, the current code as listed here seems like it is guaranteed to not work, because the OpenSSL SslProvider doesn't support FATAL_ALERT. Did you try running it?

I do believe there is a solution available by just modifying our code and maybe Netty, but it requires investigation.